src/geo.c

```diff
@@ -552,17 +609,22 @@ void georadiusGeneric(client *c, int flags) {
         return;
     }
 
+    if ((flags & GEOSEARCH) && (!(frommember || fromloc) || !(byradius || bybox))) {
+        addReplyError(c, "GEOSEARCH FROMMEMBER and FROMLOC, BYRADIUS and BYBOX "
+                         "cannot exist at the same time but one must exist");
+        return;
```

Comment:
this is a bit confusing. maybe we better split this block into two blocks (one for FROMxxx and one for BYxxx)?